x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -286,9 +288,16 @@ func (k Keeper) proofRequirementForClaim(ctx sdk.Context, claim *prooftypes.Clai
 		return requirementReason, err
 	}
 
+	proofPathSeedBlockHash, err := k.getEarliestSupplierProofCommitBlockHash(ctx, claim)
+	if err != nil {
+		return requirementReason, err
+	}
+
+	proofRequirementSeed := append(claimHash, proofPathSeedBlockHash...)
+
 	// Sample a pseudo-random value between 0 and 1 to determine if a proof is required probabilistically.
 	var randFloat float32
-	randFloat, err = poktrand.SeededFloat32(claimHash[:])
+	randFloat, err = poktrand.SeededFloat32(proofRequirementSeed)
```

Comment:
Can you move lines 291-303 into a helper and rename `randFloat` to `proofRequiredProbability`